lib/client-assets.php

```diff
@@ -463,6 +476,8 @@ function gutenberg_register_packages_styles( $styles ) {
 function gutenberg_enqueue_stored_styles( $options = array() ) {
 	$is_block_theme   = wp_is_block_theme();
 	$is_classic_theme = ! $is_block_theme;
+	// This is used to ensure that the style engine's stylesheet is loaded after the theme's stylesheet.
+	$stylesheet_handle = gutenberg_get_stylesheet_handle();
```

Comment:
I think this should get moved to after the early `return` on line 490 so we don't make a DB request for the URL unless we know we'll need it.